src/java.desktop/share/classes/javax/swing/border/EtchedBorder.java

```diff
@@ -136,15 +150,16 @@ public void paintBorder(Component c, Graphics g, int x, int y, int width, int he
 
         g.translate(x, y);
 
-        g.setColor(etchType == LOWERED? getShadowColor(c) : getHighlightColor(c));
-        g.drawRect(0, 0, w-2, h-2);
-
-        g.setColor(etchType == LOWERED? getHighlightColor(c) : getShadowColor(c));
-        g.drawLine(1, h-3, 1, 1);
-        g.drawLine(1, 1, w-3, 1);
+        // 8279614: The dark line was being overdrawn by the light line. The fix was to
+        // make sure that the dark line is always drawn second.
```

Comment:
This is a comment on the actual comment rather than whether the intent is correct.
I'm not in favour of putting bug ids into the comments. Our code will end up utterly cluttered with them
Also it is phrased to describe a fix.
So IF it were to stand as is, I would have phrased it as "what we are doing now and why", ie
// Make sure the shadow line always is drawn second because it needs to over draw the border.

Reply:
I should've posted my question separately rather than as a comment to the comment.